Source/WTF/wtf/ThreadSafeWeakPtr.h

```diff
@@ -141,7 +141,12 @@ class ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr {
 private:
     template<typename> friend class ThreadSafeWeakPtr;
     template<typename> friend class ThreadSafeWeakHashSet;
+#if COMPILER(MSVC)
+    // Workaround: error C2440: 'static_cast': cannot convert from 'WTF::ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<T,WTF::DestructionThread::Main>' to 'T &'
+    ThreadSafeWeakPtrControlBlock<T>& m_controlBlock { *new ThreadSafeWeakPtrControlBlock<T>(reinterpret_cast<T&>(*this)) };
+#else
     ThreadSafeWeakPtrControlBlock<T>& m_controlBlock { *new ThreadSafeWeakPtrControlBlock<T>(static_cast<T&>(*this)) };
+#endif
```

Comment:
This MSVC workaround made WebKit compile, but I'm not sure it's correct.  Seeing some strange crashes on the layout tests from the [win](https://ews-build.webkit.org/#/builders/10/builds/174192) bot, though even the set of tests that are crashing is inconsistent.

Waiting for layout tests to run without the patch to compare.

Reply:
Yes, I am not convinced this is correct. @achristensen07 can confirm but I believe he previously wrote a patch that updated these static_casts with reinterpret_casts and it got reverted because it introduced crashes.

Reply:
Yep.  I found a better workaround.  Trying it now (once I fix my silly typos).

Reply:
Arg.  Looking at the wrong build results.  This patch still needs a bit of work.

Reply:
The latest patch adds a `DestructionThread` template parameter (with a default value) to both `WTF::ThreadSafeWeakPtrControlBlock<>` and `WTF::ThreadSafeWeakPtrControlBlock<>` so that all three classes have consistent template parameters.

This compiles on macOS and iOS—just waiting for Win/WinCairo ports to compile.